patches/components-metrics-machine_id_provider_win.cc.patch

```diff
@@ -0,0 +1,33 @@
+diff --git a/components/metrics/machine_id_provider_win.cc b/components/metrics/machine_id_provider_win.cc
+index eb7b0d7e8fde..298f58adf23e 100644
+--- a/components/metrics/machine_id_provider_win.cc
++++ b/components/metrics/machine_id_provider_win.cc
+@@ -9,22 +9,27 @@
+ #include <winioctl.h>
+ 
+ #include "base/base_paths.h"
++#include "base/command_line.h"
+ #include "base/files/file_path.h"
+ #include "base/path_service.h"
+ #include "base/threading/scoped_blocking_call.h"
+ #include "base/win/scoped_handle.h"
++#include "brave/common/brave_switches.h"
+ 
+ namespace metrics {
+ 
+ // static
+ bool MachineIdProvider::HasId() {
+-  return true;
++  return !base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kDisableMachineId);
+ }
+ 
+ // On windows, the machine id is based on the serial number of the drive Chrome
+ // is running from.
+ // static
+ std::string MachineIdProvider::GetMachineId() {
++  if (base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kDisableMachineId)) {
```

Comment:
`GetMachineId` still gets called even if `HasId` returns false?

Reply:
Indeed, [no impl needed](https://github.com/chromium/chromium/blob/72.0.3626.28/components/metrics/metrics_state_manager.cc#L239) in `GetMachineId`.